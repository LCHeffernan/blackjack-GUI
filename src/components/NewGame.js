import React from "react";

const NewGame = () => (
  <div>
    <button
      className="new-game-button"
      data-testid="new-game-button"
      type="button"
    >
      New Game
    </button>
  </div>
);

export default NewGame;
