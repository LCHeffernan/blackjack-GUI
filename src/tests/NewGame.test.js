import React, { render, screen, fireEvent } from "@testing-library/react";
import NewGame from "../components/NewGame";

describe("NewGame", () => {
  const validProps = {
    handleNewGame: jest.fn(),
    isGameOver: true,
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <NewGame
        handleNewGame={validProps.handleNewGame}
        isGameOver={validProps.isGameOver}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
