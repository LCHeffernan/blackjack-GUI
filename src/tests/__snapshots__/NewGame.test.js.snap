// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewGame renders correctly 1`] = `
<DocumentFragment>
  <div>
    <button
      class="new-game-button"
      data-testid="new-game-button"
      type="button"
    >
      New Game
    </button>
  </div>
</DocumentFragment>
`;
